tax/errors.go

```diff
@@ -0,0 +1,30 @@
+package tax
+
+import "fmt"
+
+// Error is a general wrapper around tax errors produced during run
+// time, typically during calculations. Not to be confused with errors
+// produced from definition validation.
+type Error struct {
+	Reason Key
+}
+
+// Standard list of tax errors
+var (
+	ErrInvalidCategory      = &Error{Reason: "invalid-category"}
+	ErrInvalidRate          = &Error{Reason: "invalid-rate"}
+	ErrInvalidDate          = &Error{Reason: "invalid-date"}
+	ErrInvalidPricesInclude = &Error{Reason: "invalid-prices-include"}
+)
+
+// Error serializes the error's message.
+func (e Error) Error() string {
+	return e.Reason.String()
+}
+
+// WithMessage wraps around the original error so we can use if for matching
+// and adds a human message.
+func (e *Error) WithMessage(msg string, s ...interface{}) error {
```

Comment:
Is there any reason we can't just let the callers call `fmt.Errorf` themselves? That's a well-established convention. A special method begs the question what's different...

Reply:
I had a different approach before as the message was set on the struct. `WithMessage` means we could go back to that approach if needed and we get consistent formatting...

Reply:
YAGNI? :)

Reply:
haha true, but I still like the formatting :-)